src/metadata/EntityMetadata.ts

```diff
@@ -767,7 +767,9 @@ export class EntityMetadata {
         const entityPrefix = this.connection.options.entityPrefix;
         this.engine = this.tableMetadataArgs.engine;
         this.database = this.tableMetadataArgs.type === "entity-child" && this.parentEntityMetadata ? this.parentEntityMetadata.database : this.tableMetadataArgs.database;
-        this.schema = this.tableMetadataArgs.schema || (this.connection.options as PostgresConnectionOptions|SqlServerConnectionOptions).schema;
+        this.schema = this.tableMetadataArgs.schema
```

Comment:
better to write it multiline because its almost impossible to read this code right now